generate/genhcl/genhcl.go

```diff
@@ -181,66 +184,75 @@ func Load(
 		}
 
 		if !condition {
-			logger.Trace().Msg("condition=false, block wont be evaluated")
-
 			hcls = append(hcls, HCL{
 				label:     name,
-				origin:    loadedHCL.origin,
+				origin:    origin,
 				condition: condition,
 			})
 
 			continue
 		}
 
-		logger.Trace().Msg("evaluating block")
+		asserts := make([]config.Assert, len(hclBlock.Asserts))
+		assertsErrs := errors.L()
+		assertFailed := false
+
+		for i, assertCfg := range hclBlock.Asserts {
+			assert, err := config.EvalAssert(evalctx.Context, assertCfg)
+			if err != nil {
+				assertsErrs.Append(err)
+				continue
+			}
+			asserts[i] = assert
+			if !assert.Assertion && !assert.Warning {
+				assertFailed = true
+			}
+		}
+
+		if err := assertsErrs.AsError(); err != nil {
+			return nil, err
+		}
+
+		if assertFailed {
+			hcls = append(hcls, HCL{
+				label:     name,
+				origin:    origin,
+				condition: condition,
+				asserts:   asserts,
+			})
+			continue
+		}
 
 		gen := hclwrite.NewEmptyFile()
-		if err := copyBody(gen.Body(), loadedHCL.block.Body, evalctx); err != nil {
+		if err := copyBody(gen.Body(), hclBlock.Content.Body, evalctx); err != nil {
 			return nil, errors.E(ErrContentEval, sm, err,
 				"generate_hcl %q", name,
 			)
 		}
 
-		// TODO(i4k): filename in line below must be an absolute host path.
-		formatted, err := fmt.FormatMultiline(string(gen.Bytes()), loadedHCL.origin.String())
+		formatted, err := fmt.FormatMultiline(string(gen.Bytes()), hclBlock.Origin)
 		if err != nil {
-			// genhcl must always generate valid code that is formatable
-			// this is a severe internal error
-			logger.Error().
-				Err(err).
-				Stringer("origin", loadedHCL.origin).
-				Str("code", string(gen.Bytes())).
-				Str("label", name).
-				Msg("internal error formatting generated code")
-
 			panic(errors.E(sm, err,
-				"internal error: formatting generated code for generate_hcl %q", name,
+				"internal error: formatting generated code for generate_hcl %q:%s", name, string(gen.Bytes()),
```

Comment:
here we have a different message, maybe it would be nice to be:

```go
panic(errors.E(errors.ErrInternal, sm, err, "formatting generated code for generate_hcl %q:%s", name, string(gen.Bytes()))
```